struct_level.go

```diff
@@ -170,8 +171,33 @@ func (v *validate) ReportValidationErrors(relativeNamespace, relativeStructNames
 	for i := 0; i < len(errs); i++ {
 
 		err = errs[i].(*fieldError)
-		err.ns = string(append(append(v.ns, relativeNamespace...), err.ns...))
-		err.structNs = string(append(append(v.actualNs, relativeStructNamespace...), err.structNs...))
+
+		ns := err.ns
+		prefix := append(v.ns, relativeNamespace...)
+		if len(prefix) > 0 {
+			// Strip out type name
+			idx := strings.Index(ns, ".")
```

Comment:
Is this going to work when it's multidimensional? eg your example is `Model.Items[0].Items.Str`

What if it's two levels deep eg. `Model.Model2.Items[0].Items.Str`, this code looks like it will cut off `Model2`? perhaps this should be `LastIndex`?

Reply:
I don't think that multiple dimensions would cause any issues. Every time `Struct` gets called, it prepends the struct name. The code I've added undoes this prepending when those errors are bubbled up a level with `ReportValidationErrors`. If I understand your example correctly, that would involve multiple nested calls to `Struct`, which use `ReportValidationErrors` inside the validator function. I don't expect any problems from this, but I can write tests when I get a chance.

One thing I now realize this might break, though, is a nested call to `Var`. That wouldn't prepend a type name, so stripping out the first item from the namespace would be wrong.

Perhaps `Var` would be a way to work around my original problem, without any of these changes? I suppose I could call `Var` instead of struct, and that would produce errors without a type name attached.